src/services/utilServices/__tests__/AuthService.spec.js

```diff
@@ -101,21 +116,77 @@ describe("Auth Service", () => {
     })
   })
 
+  describe("sendOtp", () => {
+    it("should be able to send otp for whitelisted users", async () => {
+      mockUsersService.canSendEmailOtp.mockImplementationOnce(() => true)
+
+      await expect(service.sendOtp(mockEmail)).resolves.not.toThrow()
+      expect(mockUsersService.canSendEmailOtp).toHaveBeenCalledWith(mockEmail)
+      expect(mockUsersService.sendEmailOtp).toHaveBeenCalledWith(mockEmail)
+    })
+
+    it("should throw an error for non-whitelisted users", async () => {
+      mockUsersService.canSendEmailOtp.mockImplementationOnce(() => false)
+
+      await expect(service.sendOtp(mockEmail)).rejects.toThrow(AuthError)
+      expect(mockUsersService.canSendEmailOtp).toHaveBeenCalledWith(mockEmail)
+    })
+  })
+
+  describe("verifyOtp", () => {
+    const mockOtp = "123456"
+    it("should be able to verify otp, login, and return token if correct", async () => {
+      mockUsersService.verifyOtp.mockImplementationOnce(() => true)
+      jwtUtils.signToken.mockImplementationOnce(() => signedToken)
+
+      await expect(
+        service.verifyOtp({ email: mockEmail, otp: mockOtp })
+      ).resolves.toEqual(signedToken)
+      expect(mockUsersService.verifyOtp).toHaveBeenCalledWith(
+        mockEmail,
+        mockOtp
+      )
+      expect(mockUsersService.loginWithEmail).toHaveBeenCalledWith(mockEmail)
+    })
+
+    it("should throw an error if otp is incorrect", async () => {
```

Comment:
hmm, is it possible to add a test case for expired otps? (lmk if this is diff to do)

Reply:
I actually don't know if there's a special error message for expired otps - I've only been able to receive wrong otp error messages, so I don't have a test case for expired ones

Reply:
i looked through - there isn't; **all** verification is treated equally and we return the same error message even if the underlying error is different...

idt this is part of this PR (probably out of scope) but i think we should be aware of this and i'll file an issue accordingly @__@